src/map/clif.c

```diff
@@ -11484,6 +11484,25 @@ void clif_parse_ItemIdentify(int fd,struct map_session_data *sd)
 	clif_menuskill_clear(sd);
 }
 
+/// Identifying item with right-click (CZ_REQ_ONECLICK_ITEMIDENTIFY).
+/// 0A35 <index>.W
+void clif_parse_OneClick_ItemIdentify(int fd,struct map_session_data *sd)
+{
+	int cmd = RFIFOW(fd,0);
+	short idx = RFIFOW(fd, packet_db[cmd].pos[0]) - 2;
+	int n;
+
+	if ( idx < 0 )
```

Comment:
I'd add one more sanity check here: if the packet is altered so that the idx value is greater than the inventory size, we have an out-of-bounds array access a few lines below.